### PR TITLE
quic: Don't delay TCP attempt when HTTP/3 status is unknown

### PR DESCRIPTION
**Pull request by fredyw**
Commit Message: When HTTP/3 is not known to work, we should attempt both QUIC and TCP at the same time instead of giving QUIC a head start and delaying the TCP connection attempt.
Additional Description: In Envoy Mobile, the HTTP/3 status will be reset upon network change.
Risk Level: low (runtime guarded and the default is false)
Testing: unit and integration tests
Docs Changes: n/a
Release Notes: n/a
Platform Specific Features: quic, mobile
Optional Runtime guard: `envoy.reloadable_features.quic_no_tcp_delay` (default is false)

### CODE REVIEW

File: mobile/test/cc/unit/fetch_client_test.cc [hunk 1]
```diff
@@ -18,10 +18,14 @@ TEST(FetchClientTest, Http2) {
...
-  ASSERT_EQ(client.fetch({"https://www.google.com/"}, {"www.google.com"}, protocols),
+  ASSERT_EQ(client.fetch({"https://www.google.com/", "https://www.google.com/"}, {"www.google.com"},
+                         protocols),
             ENVOY_SUCCESS);
-  // TODO(fredyw): In CI, HTTP/3 does not work and will use HTTP/2 instead.
-  ASSERT_GT(protocols.front(), Http::Protocol::Http11);
+  // The first request could either be HTTP/2 or HTTP/3 because we no longer give HTTP/3 a head
+  // start.
+  ASSERT_GE(protocols.at(0), Http::Protocol::Http2);
+  // TODO(fredyw): In EngFlow CI, HTTP/3 does not work and will use HTTP/2 instead.
```
Comment (RyanTheOptimist) [inline mobile/test/cc/unit/fetch_client_test.cc:27]: Wow, that's interesting. Any idea why yet?

File: source/common/http/http_server_properties_cache_impl.cc [hunk 2]
```diff
@@ -304,6 +304,14 @@ void HttpServerPropertiesCacheImpl::resetBrokenness() {
   }
 }
 
+void HttpServerPropertiesCacheImpl::resetStatus() {
+  for (auto& protocol : protocols_) {
```
Comment (RyanTheOptimist) [inline source/common/http/http_server_properties_cache_impl.cc:308]: nit: explicit type here?

**Review by RyanTheOptimist — DISMISSED** [review]

File: mobile/test/cc/unit/fetch_client_test.cc (hunk 1 above)
Comment (fredyw) [reply mobile/test/cc/unit/fetch_client_test.cc:27]: I don't know yet. I pinged EngFlow in Slack. I suspect UDP is blocked or something even after setting https://docs.engflow.com/re/client/platform-options-reference.html#dockernetwork

File: source/common/http/http_server_properties_cache_impl.cc (hunk 2 above)
Comment (fredyw) [reply source/common/http/http_server_properties_cache_impl.cc]: Done.

**Pull request merged**